Features/849 email address

crankycoder opened the pull request:
This cleans up the transmission of email and nickname to the submit.

I've also cleaned up the code by removing the AbstractCommunicator and started to pull out pure java code into org.mozilla.mozstumbler.service.core.*

All HTTP code is now in HttpUtil and is mockable behind the IHttpUtil interface.

Email addresses sent are hashed using SHA1 and encoded as a hexdigest.

crankycoder committed:
added UI for email address and associated java code to support setting
the preference

crankycoder committed:
I changed the getNickname and getEmail methods to pure abstract methods.
Returning null doesn't do us any good and the pure abstract methods makes it abundantly clear that AbstractCommunicator really isn't an abstract class at all. We should refactor it into a delegate object that sends HTTP data.
Added getEmail method to AsyncUploader and filled out concrete implementations of getEmail, getNickname to the MLSLocationGetter

crankycoder committed:
added a call to set the email header before we invoke the uploader

crankycoder committed:
renamed the `debug_apk` target to just `debug`.  I'm lazy and don't want to type.

crankycoder committed:
Added a test and code to do a SHA1 hex digest on the email before we send it to ichnaea

crankycoder committed:
refactoring all the Zip state enum into zipper.
rename of NetworkUtils into NetworkInfo, because it's not actually doing anything, just providing information.
this is predatory work to eliminate the AbstractCommunicator

crankycoder committed:
shuffled HTTP code into org.mozilla.mozstumbler.core

crankycoder committed:
rename and clarify some variable names

crankycoder committed:
extracted a LocationAdapter to adapt JSON blobs into android Location objects

crankycoder committed:
- Extracted ILocationService interface and a concrete implementation MLS that encapsulates all communication with MLS.
- Cleaned up the Zipper utility so that exceptions are never thrown, but failure returns a null

crankycoder committed:
rewired the MLSLocationGetter to use teh ILocationService

crankycoder committed:
- finished wiring in the new MLS class into the MLSLocationGetter

crankycoder committed:
minor reworking of the AsyncTask in MLSLocationGetter for clarity

crankycoder committed:
dropped Submitter, last subclass of AbstractCommunicator
added some extra methods to check for success or error on the IReponse interface
added boolean argument to conditionally gzip compress byte content in the ILocationService methods

crankycoder committed:
Pulled AbstractCommunicator.SyncSummary up to a top level class
Deleted AbstractCommunicator

crankycoder committed:
wrapped android.util.Log with a proxy so the core package is pure java

crankycoder committed:
Merge branch 'dev' into try_merge

Conflicts:
	android/src/main/java/org/mozilla/mozstumbler/client/MainActivity.java
	android/src/main/java/org/mozilla/mozstumbler/client/Updater.java
	android/src/main/java/org/mozilla/mozstumbler/client/UploadReportsDialog.java
	android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java
	android/src/main/java/org/mozilla/mozstumbler/client/mapview/ObservationPoint.java
	android/src/main/java/org/mozilla/mozstumbler/service/uploadthread/AsyncUploader.java
	android/src/main/res/values/strings.xml
	android/src/test/java/org/mozilla/mozstumbler/client/MainDrawerActivityTest.java

crankycoder committed:
moved org.mozilla.mozstumbler.core package to org.mozilla.mozstumbler.service.core so patches to fennec are easier

crankycoder committed:
Added dependency injection of IHttpUtil to MLS
Refactored the POST method from MLS into HttpUtil
Got rid of user-agent and apikey settings in MLS, just pull directly from Prefs

crankycoder committed:
Drop ServiceTest

crankycoder committed:
Added ignore ctags to .gitignore

crankycoder committed:
added proper headers for submission of data to MLS

garvankeeley commented on android/src/main/java/org/mozilla/mozstumbler/service/uploadthread/AsyncUploader.java:
```diff
@@ -158,18 +133,24 @@ private void uploadReports(AbstractCommunicator.SyncSummary syncResult, Runnable
         try {
             DataStorageManager.ReportBatch batch = dm.getFirstBatch();
+            HashMap<String, String> headers = new HashMap<String, String>();
+            headers.put(MLS.EMAIL_HEADER, mEmail);
```
check for null email and nick

crankycoder committed:
dropped JAR files

crankycoder committed:
Well, this got rather large.

Major cleanup of AsyncUploader

* AsyncUploader no longer maintains any state anymore except for a
  single AtomicLong to hold onto the number of bytes uploaded.

* AsyncUploader requires a proper AsyncUploadParam argument when
  invoking execute() now.  This encapsulates the useWifiOnly, an
  upload listener, and the nickname
  and email address

* onPostExecute has been removed as it simply called onUploadComplete
  which is only implemented in the MetricsView, and that only invoked
  updateUIThread().  MetricsView::updateUIThread() is also invoked on
  any upload progress, so those calls have been consolidated.

* AsyncUploader::UploadSettings has been removed entirely, as it was
  just a wrapper around the useWifiOnly boolean.

* Dropped SyncSummary as we were only using it to accumulate the
  total bytes during a single execution of AsyncUploader.

* AsyncUploader::isUploading() has been removed entirely.  AsyncTasks
  can only be used once, having a singleton manage the current
  threaded upload state makes no sense.  This needs to be checked by
  subscribing to the AsyncTask using the AsyncUploaderListener
  interface.

* MetricsView manages it's own atomic isUploading flag which is
  modified when the AsyncUploader updates it's progress.

crankycoder committed:
ignore android/libs/jars

crankycoder committed:
Merge branch 'dev' into features/849-email-address

crankycoder committed:
added safety check around null email or nickname

crankycoder committed:
added MPL2 license headers

crankycoder committed:
wow i hate gradle.  pulled out updateJars into a discrete task.

crankycoder committed:
fixes as per code review

crankycoder committed:
set a default value of email and nick to be empty strings if they're set to null

crankycoder committed:
fixed some minor nits from code review
